tests/suites/test_suite_psa_crypto.function

```diff
@@ -1919,19 +1919,25 @@ void mac_bad_order( )
     /* Setup sign but try verify. */
     PSA_ASSERT( psa_mac_sign_setup( &operation, key, alg ) );
     PSA_ASSERT( psa_mac_update( &operation, input, sizeof( input ) ) );
+    TEST_ASSERT( operation.id != 0);
     TEST_EQUAL( psa_mac_verify_finish( &operation,
                                        verify_mac, sizeof( verify_mac ) ),
                 PSA_ERROR_BAD_STATE );
+    TEST_ASSERT( operation.id == 0);
```

Comment:
The rationale for this test is not obvious and warrants an explanation:

1. Multipart operations don't _have_ to abort on error, but it's something we choose to do.
2. Checking `operation.id` is a proxy for checking that the error has caused an abort.

(Note that this is a different rationale for the same check after `abort`: there, it's a consistency check on the internal state of object.)

Maybe it would be clearer (and more future-proof) to make an auxiliary macro `ASSERT_OPERATION_IS_ABORTED` or some such.

In the commit message,

>     Various functions for PSA cipher and mac operations should call abort on failure

is misleading: the API specification does not require this and we don't promise it in our implementation either. Our documentation states (e.g. for `psa_mac_update`)

> If this function returns an error status, the operation enters an error state and must be aborted by calling psa_mac_abort()

We choose to always abort, for extra robustness, and that's fine, but our internal documentation (even if it's only in the form of a commit message) should make it clear that this is a choice, and should record why we made that choice.

Reply:
I did think about having an auxiliary macro / function as you suggest - in the end I didn't as checking for id == 0 is not *exactly* checking the operation was aborted, so it's hard to think of a name that wouldn't be slightly misleading. It is tidier though, I'll add something.

Reply:
I think a short-ish macro name, plus a comment at the site of the macro definition explaining how the check relates to the property we want to test, would convey the information nicely.